### PR TITLE
Migrate to MvvmCross 6.1

### LINKED ISSUE #1861: After creating a new project, the app should direct user back to the time entry
Currently, it takes user back to the Projects view with the new project name typed in.

### LINKED ISSUE #1888: Update to MvvmCross 6.0
It's pretty cool.

### LINKED ISSUE #1957: Remove all avoidable build warnings
There are A LOT of build warnings. We should do something about it.

Some of the warnings don't have to be dealt with right now - those related to MVVMCross:
```
warning NU1701: Package 'MvvmCross.Binding 5.5.2' was restored using '.NETPortable,Version=v0.0,Profile=Profile259, .NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
```

Once we finish #1888 it should go away.

### PR DESCRIPTION
**Pull request by willsb**
Closes #1861
Closes #1888 
Closes #1957 

Needs heavy thorough testing and reviewing. Ping @simonrozsival, who wanted to see the current state of this.

Please report back on stability, since we might decide to close this is there are too many cryptic errors.

### CODE REVIEW

File: Toggl.Daneel/Startup/Setup.cs [hunk 1]
```diff
@@ -58,14 +46,20 @@ protected override IMvxNavigationService InitializeNavigationService(IMvxViewMod
...
 
-            Mvx.RegisterSingleton<IMvxNavigationService>(navigationService);
+            Mvx.RegisterSingleton(navigationService);
             return navigationService;
         }
 
+        protected override void RegisterViewTypeFinder()
+        {
+            var typeFinder = new MvxViewModelViewTypeFinder(CreateViewModelByNameLookup(), CreateViewToViewModelNaming());
+            Mvx.RegisterSingleton<IMvxViewModelTypeFinder>(typeFinder);
```
Comment (simonrozsival) [inline Toggl.Daneel/Startup/Setup.cs]: This now has to be explicit?

File: Toggl.Daneel/Startup/Setup.cs [hunk 2]
```diff
@@ -89,32 +83,34 @@ protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxAppli
...
+                   .WithUserPreferences(settingsStorage)
+                   .WithOnboardingStorage(settingsStorage)
+                   .WithNavigationService(navigationService)
+                   .WithAccessRestrictionStorage(settingsStorage)
+                   .WithPasswordManagerService<OnePasswordService>()
+                   .WithDialogService(new DialogService((ITopViewControllerProvider)Presenter))
+                   .WithErrorHandlingService(new ErrorHandlingService(navigationService, settingsStorage))
+                   .Build();
+
+            foundation.RevokeNewUserIfNeeded().Initialize();
```
Comment (simonrozsival) [inline Toggl.Daneel/Startup/Setup.cs]: 👀

File: Toggl.Giskard/Startup/Setup.cs [hunk 3]
```diff
@@ -112,9 +104,7 @@ protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxAppli
                     .WithPasswordManagerService<OnePasswordService>()
                     .WithErrorHandlingService(new ErrorHandlingService(navigationService, settingsStorage))
                     .Build();
-
-            foundation.RevokeNewUserIfNeeded().Initialize();
```
Comment (simonrozsival) [inline Toggl.Giskard/Startup/Setup.cs]: 👀

File: Toggl.Foundation.MvvmCross/ViewModels/EditDurationViewModel.cs [hunk 4]
```diff
@@ -261,7 +261,7 @@ private void onPreferencesChanged(IThreadSafePreferences preferences)
             RaisePropertyChanged(nameof(DurationFormat));
         }
 
-        public override void ViewDestroy()
+        public override void ViewDestroy(bool viewFinishing)
         {
             base.ViewDestroy();
```
Comment (daividssilverio) [inline Toggl.Foundation.MvvmCross/ViewModels/EditDurationViewModel.cs]: Should we pass the viewFinishing to `base.ViewDestroy(viewFinishing)`
What is the difference from before with this new `viewFinishing` parameter?

File: Toggl.Giskard/Views/TogglDroidTimePicker.cs [hunk 5]
```diff
@@ -6,8 +6,12 @@
 using Android.Util;
 using Android.Views;
 using Android.Widget;
+<<<<<<< HEAD
```
Comment (daividssilverio) [inline Toggl.Giskard/Views/TogglDroidTimePicker.cs]: 👀

**Comment by willsb** [comment]
So many conflicts. It's just not worth continuing this one. I'll redo it some other time.

**Pull request closed without merging**